tests/UnifiedSpecTests/Constraint/IsBsonTypeTest.php

```diff
@@ -38,9 +38,10 @@ public function testConstraint($type, $value): void
 
     public function provideTypes()
     {
-        $undefined = toPHP(fromJSON('{ "undefined": {"$undefined": true} }'));
-        $symbol = toPHP(fromJSON('{ "symbol": {"$symbol": "test"} }'));
-        $dbPointer = toPHP(fromJSON('{ "dbPointer": {"$dbPointer": {"$ref": "phongo.test", "$id" : { "$oid" : "5a2e78accd485d55b405ac12" }  }} }'));
+        $undefined = toPHP(fromJSON('{ "x": {"$undefined": true} }'))->x;
+        $symbol = toPHP(fromJSON('{ "x": {"$symbol": "test"} }'))->x;
+        $dbPointer = toPHP(fromJSON('{ "x": {"$dbPointer": {"$ref": "db.coll", "$id" : { "$oid" : "5a2e78accd485d55b405ac12" }  }} }'))->x;
+        $long = PHP_INT_SIZE == 4 ? unserialize('C:18:"MongoDB\BSON\Int64":38:{a:1:{s:7:"integer";s:10:"4294967296";}}') : 4294967296;
```

Comment:
Our tests make the best case for providing a constructor for `Int64`. Can't wait to eventually get rid of it.

Reply:
Remember that a constructor would provide users a foot-gun allowing them to force 32-bit values to serialize as 64-bit, only to have BSON decoding convert them back to basic integers. Previous discussions on this:

 * https://github.com/mongodb/mongo-php-library/issues/589
 * https://github.com/mongodb/mongo-php-driver/issues/525

Unless we implement some BSON codec API that allows users to coerce integer serialization (e.g. always use 64-bit BSON types), I think the current behavior is less likely to confuse folks while noting we can't please everyone.

Reply:
I wasn't suggesting we add a constructor - apologies if this was misunderstood. I was alluding to getting rid of Int64 if and when we drop support for 32-bit platforms. It's wishful thinking at this point, but I hope that x86 goes out of style sometime this century ^^